modules/core/nuxeo-core-storage-mem/src/main/java/org/nuxeo/ecm/core/storage/mem/MemRepositoryDescriptor.java

```diff
@@ -19,14 +19,18 @@
 package org.nuxeo.ecm.core.storage.mem;
 
 import org.nuxeo.common.xmap.annotation.XObject;
+import org.nuxeo.common.xmap.registry.XRegistry;
+import org.nuxeo.common.xmap.registry.XRegistryId;
 import org.nuxeo.ecm.core.storage.dbs.DBSRepositoryDescriptor;
 
 /**
  * Memory Repository Descriptor.
  *
  * @since 8.1
  */
-@XObject(value = "repository")
+@XObject("repository")
+@XRegistry
+@XRegistryId("@name")
```

Comment:
The `@XRegistry` annotation needs to be directly on the object for now (I did not plan to check it on parent classes on XMap processing). The `@XRegistryId` could go on the base class field though, I went back and fourth on this one and thought it might be clearer to have it on the class (but I did leave in on the base class for the similar directory use case in next PR...) --> please tell if I should change it here.

Reply:
Don't really know the complexities to handle this. My concern is about `@XRegistryId` which references a field in the base class.

Reply:
Alright then I'll move the XRegistryId.
It would not be that complex to handle the XRegistry annotation from the parent class, I just did not see any use case until now so I did not implement it (https://github.com/nuxeo/nuxeo/blob/0fc6cfcf4671a08628fe4331268c9f6e3e0e6a07/modules/runtime/nuxeo-common/src/main/java/org/nuxeo/common/xmap/XMap.java#L271 is not called on parent class)